cmd/ndm_daemonset/probe/udevprobe.go

```diff
@@ -172,7 +181,13 @@ func (up *udevProbe) scan() error {
 		}
 		newUdevice.UdevDeviceUnref()
 	}
-	up.controller.DeactivateStaleBlockDeviceResource(disksUid)
+	if !up.controller.FeatureGates.IsEnabled(features.GPTBasedUUID) {
+		up.controller.DeactivateStaleBlockDeviceResource(disksUid)
+	} else {
+		// TODO deactivate all the stale blockdevices
+		// can deactivate all the blockdevices and they will be marked as active once
+		// the event is processed.
+	}
```

Comment:
> hmm, deactivating all of them seems to be ok as if it was graceful shutdown they will be Inactive anyway. I don't see any corner case just that it will be extra etcd call for the devices that are currently attached one for making them Incative and other for making them active. What is the problem in calling DeactivateStaleBlockDeviceResource here? We can have disksUid map created for feature GPTBasedUUID) and pass it here?

@pawanpraka1  quoting your comment from #378 . 

We are generating the UUID at the last step, just before updating the resource into etcd. So, at this point after detecting a device, we don't know its UUID.

The reason for generating UUID in `eventhandler` and not here.
The `syspath` & `devpath` will serve the purpose of unique identification within the scope of the host. Only when we -need to jump into cluster wide scope, we need the actual UUID. Therefore, the UUID is created just before updating to etcd. This will also prevent us from pushing too much changes of creating partitions also, into the device detection code. 

Since UUID is not available at this point, I thought may be we could deactivate all resources belonging to this host, and then let the udev update with the details accordingly.

Reply:
hmmmm, I am fine with deactivating everything before processing anything.

Reply:
ok. then will update with the changes.

Reply:
done.